### PR TITLE
make compiler generated metadata tmpMask byte aligned

### PR DESCRIPTION
**Pull request by kamleshbhalui**

### CODE REVIEW

File: backends/dpdk/dpdkProgram.cpp [hunk 1]
```diff
@@ -286,12 +286,15 @@ void ConvertToDpdkParser::getCondVars(const IR::Expression *sv, const IR::Expres
                 "%1%, Select expression wider than 64-bit is not permitted", sv);
         return;
     }
+    auto width = sv->type->width_bits();
+    unsigned byteAlignedWidth = width + 8 - width % 8;
```
Comment (syogender) [inline backends/dpdk/dpdkProgram.cpp:290]: (width + 7) & 0x7  is the correct method
what you have done is not correct

File: testdata/p4_16_samples_outputs/psa-dpdk-errorcode-2.p4.spec [hunk 2]
```diff
@@ -59,8 +59,8 @@ struct metadata {
 	bit<16> local_metadata_data
 	bit<8> IngressParser_parser_tmp
 	bit<8> Ingress_err
-	bit<16> tmpMask
-	bit<8> tmpMask_0
+	bit<24> tmpMask
+	bit<16> tmpMask_0
```
Comment (syogender) [inline testdata/p4_16_samples_outputs/psa-dpdk-errorcode-2.p4.spec:63]: why this difference
16->24
8->16 ?

File: backends/dpdk/dpdkProgram.cpp (hunk 1 above)
Comment (psivanup) [reply backends/dpdk/dpdkProgram.cpp:290]: It should be (width + 7) & (-7).
or
((width + 7) >> 3) << 3

File: testdata/p4_16_samples_outputs/psa-dpdk-errorcode-1.p4.spec [hunk 3]
```diff
@@ -58,8 +58,8 @@ struct metadata {
 	bit<32> psa_ingress_output_metadata_egress_port
 	bit<16> local_metadata_data
 	bit<8> Ingress_err
-	bit<16> tmpMask
-	bit<8> tmpMask_0
+	bit<24> tmpMask
+	bit<16> tmpMask_0
```
Comment (psivanup) [inline testdata/p4_16_samples_outputs/psa-dpdk-errorcode-1.p4.spec:62]: This shouldn't change (after updating the formula).

**Review by psivanup — CHANGES_REQUESTED** [review]
Please check.

File: backends/dpdk/dpdkProgram.cpp (hunk 1 above)
Comment (syogender) [reply backends/dpdk/dpdkProgram.cpp:290]: > (-7).

you meant ~7

Comment (psivanup) [reply backends/dpdk/dpdkProgram.cpp:290]: yes, bitwise not to reset least three bits.

**Review by usha1830 — CHANGES_REQUESTED** [review]
Please add a test for exercising the new code. All existing tests seem to already have bytealigned bitwidth. 
Also as pointed out by Yogi and Pitchumani, the existing test output shouldn't have changed. Changing to the specified formula should fix it.

File: backends/dpdk/dpdkProgram.cpp (hunk 1 above)
Comment (kamleshbhalui) [reply backends/dpdk/dpdkProgram.cpp]: yes, thanks for catching it, it wasn't handling already aligned values.

**Review by usha1830 — APPROVED** [review]
Looks fine

**Review by psivanup — APPROVED** [review]
Looks fine.

**Pull request closed without merging**